public/paysagerenderer.js

```diff
@@ -65,20 +65,24 @@ function patchBackgroundFunctionToBeTransparentByDefault (layer) {
 }
 
 var updateObject = function (id, code) {
-  if (!canvas[id]) {
-    canvas[id] = document.createElement('canvas');
-    canvas[id].setAttribute('id', id);
-
-    document.getElementById('container').appendChild(canvas[id]);
-    console.log('canvas created for ' + id);
-  } else {
-    console.log('canvas reused for ' + id);
-    try {
-      layers[id].exit();
-    } catch (e) { }
-    delete layers[id];
+  try {
+    if (!canvas[id]) {
+      canvas[id] = document.createElement('canvas');
+      canvas[id].setAttribute('id', id);
+
+      document.getElementById('container').appendChild(canvas[id]);
+      console.log('canvas created for ' + id);
+    } else {
+      console.log('canvas reused for ' + id);
+      try {
+        layers[id].exit();
+      } catch (e) { }
+      delete layers[id];
+    }
+    layers[id] = createLayer(canvas[id], code);
+  } catch (e) {
+    console.error('Error in code object ' + id + '. Code not rendered. ' + e);
```

Comment:
I merged it as is but I think it’d look better with `, e` instead of `+ e` – it should make the stack trace clickable.